src/pykeen/models/nbase.py

```diff
@@ -328,7 +341,11 @@ def __init__(
         self,
         *,
         triples_factory: CoreTriplesFactory,
-        interaction: Hint[Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]],
+        interaction: Union[
+            str,
+            Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation],
+            Type[Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]],
```

Comment:
Is there a difference to `Hint`?

Reply:
Hint allows `None`, which we're explicitly not allowing here. There is no default interaction.

Reply:
Should we update / split `Hint` then? :sweat_smile:

Reply:
not in this PR, no. That should maybe go in class-resolver